packages/theme/middleware/checkout.js

```diff
@@ -1,31 +1,36 @@
-const cartEmpty = cart => cart.itemCount === 0
-const canEnterBilling = cart => Boolean(cart.address?.shipping) && cart.lineItems.length;
-const canEnterPayment = cart => Boolean(cart.address?.billing) && Boolean(cart.address?.shipping) && cart.lineItems.length;
+const getPathChecks = async (vsf) =>
+  await Promise.all([
+    vsf.$spree.api.getCart(),
+    vsf.$spree.api.getShipments().catch(() => null)
```

Comment:
I'm thinking about the readability of this line. To summarize - we don't care whether getShipments fails or not, but if it doesn't, we want to retrieve some data; otherwise, the data is not provided, so shipments === null

getShipments responses with 422 if a user hadn't submitted shipping details
So when cart.address.shipping is null

We could have a check for that, to await for getShipments iff cart.address.shipping exists,
 but it does not always have to be an expected state;

We wouldn't have the cart.address.shipping property on the /checkout/thank-you page, so in this approach, we should've added some checks, to verify if a page requiring getShipments to work is currently opened, but that adds some extra complexity which we probably would like to avoid.